app/models/spree/order.rb

```diff
@@ -225,7 +225,10 @@ def checkout_allowed?
     end
 
     def changes_allowed?
-      complete? && distributor&.allow_order_changes? && order_cycle&.open?
+      complete? &&
+        !shipped? &&
+        distributor&.allow_order_changes?.present? &&
+        order_cycle&.open?.present?
```

Comment:
This is an interesting use of `present?`. I didn't know that `false.present? == false`. I would have expected it to be true.

Reply:
`present?` will return `false` for "falsey" values including `false`, `nil`, empty hashes/arrays/strings, etc. I added `present?` to protect against possible nulls on these fields. This probably won't occur in reality because of database restrictions, but a null on one of these fields could make this method return `nil` instead of `false`.

Reply:
Just to clarify, I think the `present?` is added in order to enforce  `changes_allowed?` returns a boolean (not truthy/falsey) value.

If we just want to cast a possible `nil` into `false`, how about  we wrap the whole thing with a double-bang? It adds only the needed feature and is a bit cleaner. I wonder if that would work and still pass your tests?
```ruby
        !!( complete? && ...etc )
```

Reply:
That's correct, `present?` was added to enforce a boolean is returned from `changes_allowed?`. I tried your suggestion with the double-bang and it works. The code has been updated and tests are passing. Thanks for suggesting this change!